torch/utils/mkldnn.py

```diff
@@ -131,6 +131,153 @@ def forward(self, x):
             False,  # cuda_enabled
         )
 
+MKLDNN_RNN_RELU = 0
+MKLDNN_RNN_TANH = 1
+MKLDNN_LSTM = 2
+MKLDNN_GRU = 3
+
+class MkldnnLSTM(torch.jit.ScriptModule):
+    __constants__ = ['input_size', 'hidden_size', 'bias',
+                     'mode', 'num_layers', 'bidirectional', 'batch_first']
+
+    def __init__(self, dense_module):
+        super(MkldnnLSTM, self).__init__()
+
+        assert(not dense_module.training)
+        assert(not dense_module.batch_first)
+
+        self.input_size = dense_module.input_size
+        self.hidden_size = dense_module.hidden_size
+        self.bias = dense_module.bias
+        self.mode = MKLDNN_LSTM
```

Comment:
I'd rather store mode as string and remap it on the fly like here: https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/rnn.py#L140 . It's minor, but a tiny bit more backward compatible